core/crypto/src/signature.rs

```diff
@@ -404,6 +458,39 @@ impl std::fmt::Display for SecretKey {
     }
 }
 
+impl std::str::FromStr for SecretKey {
+    type Err = crate::ParseKeyError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let (key_type, key_data) = split_key_type_data(&s)?;
+        match key_type {
+            KeyType::ED25519 => {
+                let mut array = [0; ed25519_dalek::KEYPAIR_LENGTH];
+                let length = bs58::decode(key_data)
+                    .into(&mut array[..])
+                    .map_err(|err| Self::Err::InvalidData(err.to_string()))?;
+                if length != ed25519_dalek::KEYPAIR_LENGTH {
+                    return Err(Self::Err::InvalidLength(length));
+                }
+                Ok(Self::ED25519(ED25519SecretKey(array)))
+            }
+            _ => {
```

Comment:
I'd suggest to be explicit here and use `KeyType:: SECP256K1` in case we ever add another curve and then have `_ => unreachable()`

EDIT: I see you just moved it here, but still :)